### PR TITLE
[BEAM-1334] Split UsesMetrics category and tests into UsesCommittedMetrics and UsesAttemptedMetrics

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @bjchambers 
CC: @davorbonaci @amitsela

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java [hunk 1]
```diff
@@ -82,15 +84,20 @@ public void testApplyLogicalQueryNoFilter() {
...
     assertThat(results.distributions(), contains(
-        metricResult("ns1", "name1", "step1",
-            DistributionResult.create(12, 3, 3, 5),
-            DistributionResult.ZERO)));
+        committedMetricsResult("ns1", "name1", "step1", DistributionResult.create(12, 3, 3, 5))));
   }
 
+  @SuppressWarnings("unchecked")
   @Test
   public void testApplyPhysicalCountersQueryOneNamespace() {
```
Comment (bjchambers) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java]: While we're here, let's change `testApplyLogicalQueryNoFilter` to `testApplyCommittedNoFilter` and `testApplyPhysicalCountersQueryOneNamespace` to `testApplyAttemptedCountersQueryOneNamespace` (eg., `s/Logical/Committed/` and `s/Physical/Attempted`)

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java [hunk 2]
```diff
@@ -104,13 +111,21 @@ public void testApplyPhysicalCountersQueryOneNamespace() {
...
         containsInAnyOrder(
-            metricResult("ns1", "name1", "step1", 0L, 5L),
-            metricResult("ns1", "name1", "step2", 0L, 7L)));
+            committedMetricsResult("ns1", "name1", "step1", 0L),
+            committedMetricsResult("ns1", "name1", "step2", 0L)));
   }
 
+  @SuppressWarnings("unchecked")
   @Test
   public void testApplyPhysicalQueryCompositeScope() {
```
Comment (bjchambers) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java]: same re: naming

**Comment by aviemzur** [comment]
@bjchambers 

Applied the renames you requested.

**Pull request closed without merging**